src/main/java/com/baidu/hugegraph/rest/RestClient.java

```
@@ -90,9 +94,70 @@ public RestClient(String url, String user, String password, int timeout,
+                            throws CertificateException {
+                    }
+
+                    @Override
+                    public X509Certificate[] getAcceptedIssuers() {
+                        return null;
+                    }
+                }
+        };
+    }
+
     public RestClient(String url, ClientConfig config) {
-        this.client = ClientBuilder.newClient(config);
+        Client client = null;
+        Object protocol = config.getProperties().get("protocol");
+        if (protocol != null && protocol.equals("https")) {
+            try {
+                SslConfigurator sslConfig = SslConfigurator.newInstance();
+                sslConfig.trustStoreFile(config.getProperties().get("trustStoreFile").toString())
+                        .trustStorePassword(config.getProperties().get("trustStorePassword").toString());
+                sslConfig.securityProtocol("SSL");
+                SSLContext sc = sslConfig.createSSLContext();
+                TrustManager[] trustAllCerts = getTrustManager();
+                sc.init(null, trustAllCerts, new java.security.SecureRandom());
+
+                client = ClientBuilder.newBuilder()
+                        .hostnameVerifier(new HostnameVerifier() {
+                            @Override
+                            public boolean verify(String hostname, SSLSession session) {
+                                if (!url.equals("") && url.contains(hostname)) {
+                                    return true;
+                                } else {
+                                    HostnameVerifier hv = HttpsURLConnection.getDefaultHostnameVerifier();
+                                    return hv.verify(hostname, session);
+                                }
+                            }
+                        })
+                        .sslContext(sc)
+                        .build();
+            } catch (Exception e) {
+                e.printStackTrace();
```

don't swallow exception